storagepool/metadataBasedHealthCheck.go

```diff
@@ -0,0 +1,75 @@
+package storagepool
+
+import (
+	"time"
+
+	"github.com/rancher/go-rancher-metadata/metadata"
+)
+
+type metadataBasedHealthCheck struct {
+	version     string
+	prevHosts   map[string]string
+	metadataUrl string
+}
+
+func (mt *metadataBasedHealthCheck) populateHostMap() (map[string]string, error) {
+	m := metadata.NewHandler(mt.metadataUrl)
+
+	version, err := m.GetVersion()
+	if err != nil {
+		return nil, err
+	}
+	if version == mt.version {
+		return mt.prevHosts, nil
+	} else {
+		mt.version = version
+	}
+
+	activeHosts := map[string]string{}
+	timeStamp := time.Now().Format(time.RFC1123Z)
+	stack, err := m.GetSelfStack()
+	if err != nil {
+		return nil, err
+	}
+	stackName := stack.Name
+	serviceNames := stack.Services
+
+	services, err := m.GetServices()
+	if err != nil {
+		return nil, err
+	}
+
+	possibleContainers := []string{}
+
+	for _, service := range services {
+		if contains(serviceNames, service.Name) && service.StackName == stackName {
```

Comment:
`contains(serviceNames, service.Name)` is redundant, `service.StackName == stackName` is sufficient